paasta_tools/setup_kubernetes_job.py

```diff
@@ -142,12 +142,13 @@ def setup_kube_deployments(
                 app.kube_deployment.instance,
             ) not in existing_apps:
                 app.create(kube_client)
-            elif (
-                app.kube_deployment not in existing_kube_deployments
-                or autoscaling_is_paused()
-            ):
+            elif app.kube_deployment not in existing_kube_deployments:
```

Comment:
Can you add some comments explaining what the logic is doing here?  It's a bit complicated to understand what's happening just by looking at the code.

Secondary question: can any of the cases where we call `app.update` be condensed?  That might make it easier to understand?  Or it might make it harder/more complicated, not sure.

Reply:
Hmm, I thought about different condensed version of this, but that seems to make things very confusing. Having it explicitly laid out seems like the easiest way to comprehend.